hal/src/nRF52840/device_code.cpp

```diff
@@ -38,29 +38,28 @@ const auto DEVICE_NAME_DCT_OFFSET = DCT_SSID_PREFIX_OFFSET;
 
 const auto DEVICE_NAME_MAX_SIZE = DEVICE_NAME_DCT_SIZE - 1;
 
-const auto SETUP_CODE_SIZE = DCT_DEVICE_CODE_SIZE;
 const auto SETUP_CODE_DCT_OFFSET = DCT_DEVICE_CODE_OFFSET;
 
-int getDeviceSetupCode(char* code) {
+} // ::
+
+int get_device_setup_code(char* code, size_t size) {
     // Check if the device setup code is initialized in the DCT
-    int ret = dct_read_app_data_copy(SETUP_CODE_DCT_OFFSET, code, SETUP_CODE_SIZE);
-    if (ret < 0 || !isPrintable(code, SETUP_CODE_SIZE)) {
+    int ret = dct_read_app_data_copy(SETUP_CODE_DCT_OFFSET, code, size);
```

Comment:
Why the change to use `size` instead of fixed constant? If buffer provided is larger than `SETUP_CODE_SIZE` we'll read garbage data. The existing code was saner.